webpack.config.babel.js

```diff
@@ -0,0 +1,100 @@
+import path from 'path';
+import webpack from 'webpack';
+
+let config = {
+	entry: {
+		// admin ui bundle
+		// admin: path.resolve(__dirname, './admin/client/index'),
+		// fields/filters bundle
+		// fields: [
+		// 	path.resolve(__dirname, './admin/client/fields'),
+		// 	path.resolve(__dirname, './admin/client/filters'),
+		// ],
+		// packages bundle
+		admin: `${__dirname}/admin/client/App`,
+		signin: `${__dirname}/admin/client/Signin`,
+	},
+	output: {
+		filename: '[name].js',
+		path: path.resolve(__dirname, './admin/public/js'),
+		// NOTE: hardcoding keystone
+		publicPath: '/keystone/js/',
+		pathinfo: true,
+	},
+	module: {
+		loaders: [
+			{
+				test: /fields\/.*(Field|Column|Filter)\.jsx?$/i,
+				loader: 'react-proxy',
```

Comment:
It wraps a react component with a `require.ensure` wrapper. Since React doesn't require synchronous loading, that works great. Net result is that each field loads its own dependencies on the fly, like CodeMirror.